cinema/views.py

```diff
@@ -56,3 +95,61 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if self.action in ("retrieve", "list"):
+            queryset = (
+                queryset
+                .select_related()
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__rows")
+                        * F("cinema_hall__seats_in_row")) - Count("tickets")
+                )
+            )
+
+        if date:
+            date = datetime.strptime(date, "%Y-%m-%d")
+            queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            movie = int(movie)
+            queryset = queryset.filter(movie__id=movie)
+
+        return queryset.distinct()
+
+
+class OrderSetPagination(PageNumberPagination):
+    page_size = 2
+    page_size_query_param = "page_size"
+    max_page_size = 20
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderSetPagination
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return OrderListSerializer
+        return OrderSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset.filter(user=self.request.user)
+        if self.action == "list":
+            return (queryset.
```

Comment:
style looks odd, and parens are otherwise unnecessary. I'd recommend using autoformatters (like `black`) to solve this issues seamlessly